client/src/actions/authActions.js

```diff
@@ -1,18 +1,32 @@
-import { AUTH_SUCCESS, AUTH_LOGOUT } from "constants/action-types";
-import { getAuthToken, removeAuthToken } from "utils/auth-token";
+import axios from "axios";
+import {
+  AUTH_LOGOUT,
+  AUTH_LOGOUT_ERROR,
+  SET_USER,
+} from "constants/action-types";
 
-// Note: for production apps, both localstorage & cookies contain risks to store user & auth data
-export const initAuth = () => {
-  return (dispatch) => {
-    const token = getAuthToken();
-
-    if (token) {
-      dispatch({ type: AUTH_SUCCESS, payload: { token } });
-    }
+// Token stored in httpOnly cookie set/cleared by server
+export const initAuth =  () => {
+  return async (dispatch) => {
+    try {
+      const { data: user } = await axios.get("/api/users/current");
+      dispatch({ type: SET_USER, payload: { user } });
+    } catch {} // TODO: consider are there any errors to handle?
```

Comment:
well request could always go wrong so might be good to display an error
we don't really have more application-level error display in the UI I think
we could use some kind of toast / popup message which would be easy to implement everywhere
the least you could do is log the error here to the ocnsole

Reply:
There is an error with no cookie (logged out) - 401 Unauthorized - but this is expected here so I don't want to do anything about. But maybe handle other errors a certain way? Maybe on the backend if no cookie we return success with {user: null} because it isn't an actual error so it's handled similar to this video on front-end: https://www.youtube.com/watch?v=jsaOTcBe-dM

Reply:
I updated b/e to send null if no authentication. So the errors should only be for unexpected DB/server things (just logged for now).